totp.go

```diff
@@ -38,13 +42,15 @@ func BarcodeImage(label string, secretkey []byte, opt *Options) ([]byte, error)
 
 	u.RawQuery = params.Encode()
 
-	c, err := qr.Encode(u.String(), qr.M)
+	c, err := qr.Encode(u.String(), qr.ECLevelM)
 
 	if err != nil {
 		return nil, err
 	}
 
-	return c.PNG(), nil
+	buf := bytes.NewBuffer([]byte{})
+	err = png.Encode(buf, c.Image(8))
```

Comment:
I don't know if this is possible, but if png.Encode writes a partial result to the buffer, then returns an error, then we'll return a non-nil buffer and a non-nil error. To preserve the previous behavior (or atleast to make it obvious that this can't happen), can you add another error check here?